server/crawlers/nytCrawler.js

```diff
@@ -0,0 +1,73 @@
+const axios = require('axios')
+
+async function nytCrawler(companyName) {
+  try {
+
+    //formatting company name in case there's spaces
+    let formattedName = ''
+    for (let letter of companyName) {
+      if (letter === " ") {
+        continue
+      }
+      else formattedName += letter
+    }
+
+    //setting dates for one week prev and today to fit nyt api convention
+    let oneWeekPrev = new Date()
+    let pastDate = oneWeekPrev.getDate() - 7;
+    oneWeekPrev.setDate(pastDate)
+    const prevYear = oneWeekPrev.getFullYear().toString()
+    let convertedPrevMonth = (oneWeekPrev.getMonth() + 1).toString()
+    if (convertedPrevMonth.length === 1) {
+      convertedPrevMonth = `0${convertedPrevMonth}`
+    }
+    let convertedPrevDay = oneWeekPrev.getDate().toString()
+    if (convertedPrevDay.length === 1) {
+      convertedPrevDay = `0${convertedPrevDay}`
+    }
+    let formattedPrevDate = prevYear + convertedPrevMonth + convertedPrevDay
+
+    let today = new Date()
+    const currentYear = today.getFullYear().toString()
+    let convertedCurrMonth = (today.getMonth() + 1).toString()
+    if (convertedCurrMonth.length === 1) {
+      convertedCurrMonth = `0${convertedCurrMonth}`
+    }
+    let convertedCurrDay = today.getDate().toString()
+    if (convertedCurrDay.length === 1) {
+      convertedCurrDay = `0${convertedCurrDay}`
+    }
+    let formattedCurrDate = currentYear + convertedCurrMonth + convertedCurrDay
```

Comment:
Everything is looking good to me! One small suggestion I would like to make though is consider looking into the library moment.js: it makes handling dates sooooooo much easier. I also plan on using it with the news api. 

Heres a snippet I'm using to get one month ago from today formatted in YYYY-DD-MM for example:
```
const lastMonth = moment().subtract(1, "months").format("YYYY-DD-MM");```

Reply:
just refactored, ty for the tip @grogsy